tokio-util/src/lib.rs

```diff
@@ -17,7 +17,20 @@
 //! [`tokio`] crate. However, `tokio-util` _will_ respect Rust's
 //! semantic versioning policy, especially with regard to breaking changes.
 //!
+//! # Cancellation safety
+//!
+//! When using `tokio::select!` in a loop to receive messages from multiple sources,
+//! you should make sure that the receive call is cancellation safe to avoid
+//! losing messages. This section goes through various common methods and
+//! describes whether they are cancel safe.  The lists in this section are not
+//! exhaustive.
+//!
+//! * [`futures_util::sink::SinkExt::send`]: if send is used as the event in a
+//! `tokio::select!` statement and some other branch completes first, then it is
+//! guaranteed that the message was not sent, but the message itself is lost.
```

Comment:
I think I would rather put a section in the docs for `Framed`, `FramedRead`, and `FramedWrite`.

Reply:
Should I add the doc to these as well?
* `CopyToBytes`
* `StreamReader`
* `UdpFramed`
* `PollSender`
These implement `Sink` as well.